python/lbann/launcher/__init__.py

```diff
@@ -13,6 +13,7 @@
 
 def run(trainer, model, data_reader, optimizer,
         work_dir=None,
+        proto_file_name='experiment.prototext',
```

Comment:
If I'm honest, I don't think modifying this function is the right way to go. It's supposed to be a convenience driver, and adding more knobs to handle multiple models will make it bloated, convoluted, and more likely to fail when a user puts in a weird combination of options.

I would prefer if we wrote a function built off of `make_batch_script` and put in an application dir. This function would take in multiple models and save each one of them with `save_proto`. It would then construct a batch script with the correct LBANN invocation to specify multiple models.

Reply:
I believe we discussed this at a previous LBANN meeting. This does no harm IMO and it is just one line change instead of having user write the function you suggested for every application. I agree with that the _run_ function has a long list of arguments already but I don't think this one argument will do any harm; it is optional but also provide users flexibility to rename the default prototext filename if so desire.

Reply:
From what I gather, we require many more changes for your use-case. You need to save multiple models, each with a different name. This function is still only capable of saving a single model, although now you can change the filename. If `run` won't work for your eventual use-case, we might as well transition to a new launcher sooner rather than later.